### ISSUE TITLE
Mailresponse to customer not working at all after editing a "Trigger" event text

### ISSUE THREAD

**Issue by jacotec** [zammad/zammad#385]
I'm editing the texts in the "Trigger" section to create my own replies to the customers i.e. when a new ticket is created. I did not change the trigger properties itself, just the text in the body. I'm keeping the variables as they are, just editing the surrounding text.

Example:

I've changed the ticket receive notification to the customer as below:

![image](https://cloud.githubusercontent.com/assets/3321655/20140274/925c4dbe-a68a-11e6-9437-9eb8b326da0c.png)

Now the system stops sending any notification to the customer when a new ticket is created! To check that really no variables got broken, I've reduced the message to just "Ihre Anfrage" in the subject and "Test" in the body - still the message is not sent.

I'd appreciate information under what circumstances the system just "ignores" an edited text, resulting in not sending this message anymore. How can I avoid this? And how can I debug this?

**Comment by jacotec** [comment]
OK, the mail template killed the whole system (see #389, #214 ).

I've some ideas for an improvement:

- Syntax check templates before saving
- Make errors in templates "uncritical" (maybe variables don't work if they are wrong and are misdisplayed, but it never should kill the whole system)

@martini : Any idea what's the issue with my edited text?

> Ihre Anfrage (#{config.ticket_hook}#{ticket.number}) wurde empfangen und wird von unserem Team bearbeitet.
> 
> Wenn Sie weitere Nachrichten zu diesem Vorgang hinzufügen möchten, antworten Sie bitte einfach auf diese eMail, oder bearbeiten Sie sich den Verlauf direkt im Helpdesk-System unter dem Link: #{config.http_type}://#{config.fqdn}/#ticket/zoom/#{ticket.id}
> 
> Falls dies Ihr erster Kontakt mit unserem Helpdesk-System ist, müssen Sie sich einmalig ein Paßwort erstellen. Anschließend können Sie alle Ihre Anfragen einsehen und bearbeiten. Klicken Sie bitte hierzu auf diesen Link:
> https://helpdesk.mydomain.de/#password_reset
> 
> Ihr #{config.product_name}-Team

**Comment by jacotec** [comment]
This might help:

I've created a new trigger notification, pasted exactly the same text from the old, non-working and deactivated notification into it (I did a perfect 1:1 copy) and this one works perfectly now.

@martini : I've kept the "broken" one deactivated in case I can help you with some logging data / database dump to resolve your blocking bug ASAP. Just tell me if this is not needed and I'm good to delete it.

**Comment by martini** [comment]
@jacotec I found the problem. Just for me, what browser do you use?

**Comment by jacotec** [comment]
@martini : Mostly Firefox, but sometimes Chrome. I rendered another text one unusable today :-( , I did these changes on Chrome. But the same happened on my test VM and that was definitely Firefox.

**Comment by martini** [comment]
@jacotec Unfortunately I was not able to get it reproduced like you reported.

Can you search for the trigger and add the output to this bug by using ($ID is 1,2,...):

`rails r 'p Trigger.find($ID).inspect'`

Thanks!

**Comment by jacotec** [comment]
@martini Here is the output of the "broken" (and now inactive) ticket which has caused the mess:

`irb(main):001:0> Trigger.find(1).inspect    
=> "#<Trigger id: 1, name: \"KundenbestÃ¤tigung neues Ticket\", condition: {\"ticket.action\"=>{\"operator\"=>\"is\", \"value\"=>\"create\"}, \"ticket.state_id\"=>{\"operator\"=>\"is not\", \"value\"=>\"4\"}, \"article.type_id\"=>{\"operator\"=>\"is\", \"value\"=>[\"1\", \"5\", \"11\"]}, \"article.sender_id\"=>{\"operator\"=>\"is\", \"value\"=>\"2\"}}, perform: {\"notification.email\"=>{\"body\"=>\"<div><div>Ihre Anfrage (\\\#{config.ticket_hook}\\\#{ticket.number}) wurde empfangen und wird von unserem Team bearbeitet.</div><div><br></div><div>Wenn Sie weitere Nachrichten zu diesem Vorgang hinzufÃ¼gen mÃ¶chten, antworten Sie bitte einfach auf diese eMail, oder bearbeiten Sie sich den Verlauf direkt im Helpdesk-System unter dem Link: \\\#{config.http_type}://\\\#{config.fqdn}/#ticket/zoom/\\\#{<a href=\\\"http://ticket.id\\\" title=\\\"http://ticket.id\\\" target=\\\"_blank\\\">ticket.id</a>}</div><div><br></div><div>Falls dies Ihr erster Kontakt mit unserem Helpdesk-System ist, mÃ¼ssen Sie sich einmalig ein PaÃwort erstellen. AnschlieÃend kÃ¶nnen Sie alle Ihre Anfragen einsehen und bearbeiten. Klicken Sie bitte hierzu auf diesen Link:</div><div><a href=\\\"https://helpdesk.fast-break.de/#password_reset\\\" title=\\\"https://helpdesk.fast-break.de/#password_reset\\\" target=\\\"_blank\\\">https://helpdesk.fast-break.de/#password_reset</a></div><div><br></div><div>Ihr \\\#{config.product_name}-Team</div></div>\\n<div><br></div><div><br></div>\", \"recipient\"=>\"ticket_customer\", \"subject\"=>\"Ihre Anfrage\"}}, disable_notification: true, note: nil, active: false, updated_by_id: 3, created_by_id: 1, created_at: \"2016-11-09 11:46:44\", updated_at: \"2016-11-09 20:18:54\">"`

**Comment by thorsteneckel** [comment]
Hi @jacotec - we debugged and found the issue that is responsible for this and #389 (wich is a sub issue). Please see the commit above for the fix. we added your case as a new test. This fix will be in the next release of Zammad on monday or (in the next couple of minutes) in our devel branch / packager.io repo. I'll close this one here. Feel free to reopen and/or leave feedback. Thanks for your support!

**Comment by jacotec** [comment]
Hi @thorsteneckel - sounds great :-)

Can you have a look on my comment in #392 please? I've changed my repo to the dev branch, but "yum update" does not do anything, it mentions it's still the same version as the 1.0 release. What's wrong here?

**Comment by monotek** [comment]
Ah. Is see the problem. Develop branch is still tagged 0.1.8. you have to
install it with rpm -i Zammad.... After downloading it...

Marco Jakobs <notifications@github.com> schrieb am Fr., 11. Nov. 2016,
18:04:

**Comment by jacotec** [comment]
OK, I'll try that ... hopefully this will "install over".
I did a "yum remove Zammad", then changed to the dev branch and did "yum install Zammad". This flattens the file system, but I was able to reconnect to my database and my stuff was still there. Not sure what happens to attachments as they are in the file system.

Of course I did a snapshot of the ESXi-VM before ... will revert and try the RPM method, see if this will work without a complete uninstall ...
